main.go

```diff
@@ -0,0 +1,28 @@
+package main
```

Comment:
I thought the server package was meant to contain the reusable components of the server, and that the server executable was going to be built elsewhere. But I do like the idea of providing an executable as well. 

Maybe we should separate these components out a bit more? E.g. we could move all reusable protocol implementation code to a separate package (maybe ``protocol``), and keep the server package for the reference implementation?